src/main/java/org/opensearch/flowframework/workflow/CreateIndexStep.java

```diff
@@ -117,6 +117,16 @@ public String getName() {
         return NAME;
     }
 
+    // TODO : Move to index management class, pending implementation
+    /**
+     * Checks if the given index exists
+     * @param indexName the name of the index
+     * @return boolean indicating the existence of an index
+     */
+    public boolean doesIndexExist(String indexName) {
```

Comment:
I dont think we'll need to pass the cluster service around for these utility methods if we have something like the `IndexManagement` class of AD ([link](https://github.com/opensearch-project/anomaly-detection/blob/040cd7e33f8e696ffe7748140c9b0ecc7475d513/src/main/java/org/opensearch/timeseries/indices/IndexManagement.java#L141)). 

I think @amitgalitz is working on separating these common index utility methods out of this create index step so that he can use them for his state index implementation